sdk/python/kfp/_client.py

```diff
@@ -188,8 +194,7 @@ def __init__(self,
             api_client)
         self._healthz_api = kfp_server_api.api.healthz_service_api.HealthzServiceApi(
```

Comment:
Actually, `self.get_kfp_healthz()` relies on `self._healthz_api` being initialized first. If you move `self.get_kfp_healthz()` up, it will never succeed.

Reply:
True that. Made logic simpler. 

I basically set the namespace if user has explicitly called out namespace. I only ovewrite current instance though to avoid having side effects and changing the default which is written in the file.